pkg/snowflake/db.go

```diff
@@ -282,3 +282,79 @@ func (db *DB) RecreateTableOnMaterializationTypeMismatch(ctx context.Context, as
 
 	return nil
 }
+
+func (db *DB) PushColumnDescriptions(ctx context.Context, asset *pipeline.Asset) error {
+	tableComponents := strings.Split(asset.Name, ".")
+	var schemaName string
+	var tableName string
+	switch len(tableComponents) {
+	case 2:
+		schemaName = strings.ToUpper(tableComponents[0])
+		tableName = strings.ToUpper(tableComponents[1])
+	case 3:
+		schemaName = strings.ToUpper(tableComponents[1])
+		tableName = strings.ToUpper(tableComponents[2])
+	default:
+		return nil
+	}
+	anyColumnHasDescription := false
+	colsByName := make(map[string]*pipeline.Column, len(asset.Columns))
+
+	for _, col := range asset.Columns {
+		colsByName[col.Name] = &col
+		if col.Description != "" {
+			anyColumnHasDescription = true
+		}
+	}
+
+	if asset.Description == "" && (len(asset.Columns) == 0 || !anyColumnHasDescription) {
+		return errors.New("no metadata to push: table and columns have no descriptions")
+	}
```

Comment:
I think the `anyColumnHasDescription` is not necessary anymore, given that we've put this behind a flag anyway.
```suggestion
	colsByName := make(map[string]*pipeline.Column, len(asset.Columns))
	for _, col := range asset.Columns {
		colsByName[col.Name] = &col
	}

	if asset.Description == "" && (len(asset.Columns) == 0) {
		return errors.New("no metadata to push: table and columns have no descriptions")
	}
```